components/nifosg/controller.hpp

```diff
@@ -48,13 +48,14 @@ namespace NifOsg
     {
     public:
         typedef typename MapT::ValueType ValueT;
+        typedef typename MapT::MapType InnerMapType;
 
         ValueInterpolator()
             : mDefaultVal(ValueT())
         {
         }
 
-        ValueInterpolator(boost::shared_ptr<const MapT> keys, ValueT defaultVal = ValueT())
```

Comment:
The const is there because in certain cases the same keys might be shared between several interpolators/controllers, so we don't want to modify the keys.

Not a big deal removing the const, we can declare the same intent by adding a comment.